// data-analysis-python\src\get_jp_stock_info\get_jp_stock_info.proto

syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/get_jp_stock_info";

// 日本株の全フィールドを取得するためのサービスを定義
service GetJpStockInfoService {
  rpc GetJpStockInfo (GetJpStockInfoRequest) returns (GetJpStockInfoResponse);
}

// リクエストメッセージ（パラメータが不要の場合でも定義）
message GetJpStockInfoRequest {}

// 株式情報を格納するメッセージ
message StockInfo {
  string country = 1;
  string symbol = 2;
  string name = 3;
  string full_name = 4;
  string isin = 5;
  string currency = 6;
  string stock_exchange = 7;
  string sector = 8;
  string industry = 9;
}

// 株式情報を格納するレスポンスメッセージ
message GetJpStockInfoResponse {
  repeated StockInfo stocks = 1;  // 株式情報のリスト
}

// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_jp_stock_info --python_out=src/get_jp_stock_info --grpc_python_out=src/get_jp_stock_info src/get_jp_stock_info/get_jp_stock_info.proto

// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_jp_stock_info --go_out=./src/service/ms_gateway/get_jp_stock_info --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_jp_stock_info --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_jp_stock_info/get_jp_stock_info.proto
